src/Data/Mapper/PositionMapper.php

```diff
@@ -27,4 +27,24 @@ public function getPositions($options = [])
 
         return [$this->mapObjects($result), $paginate];
     }
+
+    protected function mapObjects(array $records)
+    {
+        $results = [];
+        foreach ($records as $record) {
+            $entity = new $this->entity;
```

Comment:
What if we still stored the data in the `hash`, but avoided all magic getters and setters? We could create generic `get` and `set` methods that would allow us to create:

```php
public function getAccountId()
{
    return $this->get('account_id');
}
```

Then we could still do this:
```php
public function getData()
{
    return $this->data;
}
```
Just a thought moving forward. I do like the ability to make it very understandable, but the idea of writing `new_field` in so many different places scares me a little.